pkg/codegen/utilities.go

```diff
@@ -113,16 +112,13 @@ func (s Set) Has(v interface{}) bool {
 	return ok
 }
 
-// SortedKeys returns a sorted list of keys for the given map. The map's key type must be of kind string.
-func SortedKeys(m interface{}) []string {
-	mv := reflect.ValueOf(m)
-
-	contract.Require(mv.Type().Kind() == reflect.Map, "m")
-	contract.Require(mv.Type().Key().Kind() == reflect.String, "m")
-
-	keys := make([]string, mv.Len())
-	for i, k := range mv.MapKeys() {
-		keys[i] = k.String()
+// SortedKeys returns a sorted list of keys for the given map.
+func SortedKeys[T any](m map[string]T) []string {
+	keys := make([]string, len(m))
+	var i int
+	for k := range m {
+		keys[i] = k
+		i++
```

Comment:
If we're okay taking a dependency on golang.org/x/exp, this can be shortened and generalized to:

```
import "golang.org/x/exp/maps"
import "golang.org/x/exp/slices"
import "golang.org/x/exp/constraints"

func SortedKeys[K constraints.Ordered, V any](m map[K]V) []K {
  keys := maps.Keys(m)
  slices.Sort(keys)
  return keys
}
```

The "maps" and "slices" packages have both been accepted for the standard library.

Reply:
Think I'll err on keeping this a small change for now, not sure we want to pull in exp. Maybe something to talk about next ideation, probably a lot of scope for using it giving all the set and sorting stuff we do.